src/vs/workbench/api/node/extHostLanguageFeatures.ts

```diff
@@ -1003,14 +1023,18 @@ export class ExtHostLanguageFeatures implements ExtHostLanguageFeaturesShape {
 
 	registerRenameProvider(selector: vscode.DocumentSelector, provider: vscode.RenameProvider): vscode.Disposable {
 		const handle = this._addNewAdapter(new RenameAdapter(this._documents, provider));
-		this._proxy.$registerRenameSupport(handle, selector);
+		this._proxy.$registerRenameSupport(handle, selector, RenameAdapter.supportsResolving(provider));
 		return this._createDisposable(handle);
 	}
 
 	$provideRenameEdits(handle: number, resource: UriComponents, position: IPosition, newName: string): TPromise<modes.WorkspaceEdit> {
 		return this._withAdapter(handle, RenameAdapter, adapter => adapter.provideRenameEdits(URI.revive(resource), position, newName));
 	}
 
+	$resolveInitialRenameValue(handle: number, resource: URI, position: IPosition): TPromise<modes.RenameInitialValue> {
```

Comment:
The `URI` argument won't be of that type at runtime. That's because the function will be called by our JSON-rpc internals and the runtime type will be `UriComponents` (from which you then create an `URI` using `revive`).